sycl/test/extensions/free_function_kernels/free_function_kernels_diagnostics.cpp

```diff
@@ -13,13 +13,13 @@ void ndRangeKernelReference(int &Ref) {}
 
 // Diagnostic for these violations of the restrictions haven't been implemented
 // yet.
-// TODO: Add expected error when it will be implemented.
 SYCL_EXT_ONEAPI_FUNCTION_PROPERTY(syclexp::single_task_kernel)
-void singleTaskKernelVariadic(...) {}
+void singleTaskKernelVariadic(...) {
+} // expected-error {{free function kernel cannot be a variadic function}}
```

Comment:
I think we should not have tests verifying diagnostics in sycl/ directory. Instead, we need to place them in clang/test/SemaSYCL. I think these two cases that are added here in this patch can be removed since they are already checked in clang/test/SemaSYCL. We can leave the test file for now to capture which restrictions are not yet diagnosed properly.
```suggestion
```